plugins/inputs/aos/aos.go

```diff
@@ -336,261 +402,351 @@ func (ssl *StreamAos) MsgReader(r io.Reader) {
-				myAlertData := newAlert.GetRouteAlert()
-				ssl.ExtractAlertData(alertTypeName, tags, myAlertData, raise)
-			case "liveness_alert":
-				myAlertData := newAlert.GetLivenessAlert()
-				ssl.ExtractAlertData(alertTypeName, tags, myAlertData, raise)
-			case "deployment_alert":
-				myAlertData := newAlert.GetDeploymentAlert()
-				ssl.ExtractAlertData(alertTypeName, tags, myAlertData, raise)
-			case "blueprint_rendering_alert":
-				myAlertData := newAlert.GetBlueprintRenderingAlert()
-				ssl.ExtractAlertData(alertTypeName, tags, myAlertData, raise)
-			case "counters_alert":
-				myAlertData := newAlert.GetCountersAlert()
-				ssl.ExtractAlertData(alertTypeName, tags, myAlertData, raise)
-			case "mac_alert":
-				myAlertData := newAlert.GetMacAlert()
-				ssl.ExtractAlertData(alertTypeName, tags, myAlertData, raise)
-			case "arp_alert":
-				myAlertData := newAlert.GetArpAlert()
-				ssl.ExtractAlertData(alertTypeName, tags, myAlertData, raise)
-			case "headroom_alert":
-				myAlertData := newAlert.GetHeadroomAlert()
-				ssl.ExtractAlertData(alertTypeName, tags, myAlertData, raise)
-			case "lag_alert":
-				myAlertData := newAlert.GetLagAlert()
-				ssl.ExtractAlertData(alertTypeName, tags, myAlertData, raise)
-			case "mlag_alert":
-				myAlertData := newAlert.GetMlagAlert()
-				ssl.ExtractAlertData(alertTypeName, tags, myAlertData, raise)
-			case "probe_alert":
-				myAlertData := newAlert.GetProbeAlert()
-				ssl.ExtractAlertData(alertTypeName, tags, myAlertData, raise)
-			case "config_mismatch_alert":
-				myAlertData := newAlert.GetConfigMismatchAlert()
-				ssl.ExtractAlertData(alertTypeName, tags, myAlertData, raise)
-			case "extensible_alert":
-				myAlertData := newAlert.GetExtensibleAlert()
-				ssl.ExtractAlertData(alertTypeName, tags, myAlertData, raise)
-			case "test_alert":
-				myAlertData := newAlert.GetTestAlert()
-				ssl.ExtractAlertData(alertTypeName, tags, myAlertData, raise)
-
-			default:
-				log.Printf("W! Alert Type - %s, Not Supported Yet", alertTypeName)
+	if newAlert != nil {
+		if ssl.IsSequencedStream {
+			ssl.alertSeqInit.Do(func() {
+				ssl.AlertSequence = sequencedMsg.GetSeqNum()
+				ssl.AlertSeqGap = 0
+				ssl.reportMessageLoss("alert", 0, 0)
+			})
+			seqGap := sequencedMsg.GetSeqNum() - ssl.AlertSequence
+			if seqGap != ssl.AlertSeqGap {
+				log.Printf("W! Alert sequence number expected %d got %d", ssl.AlertSequence, sequencedMsg.GetSeqNum())
+				ssl.reportMessageLoss("alert", ssl.AlertSequence, sequencedMsg.GetSeqNum())
+				ssl.AlertSequence = sequencedMsg.GetSeqNum() + 1
+				ssl.AlertSeqGap = seqGap
+			} else {
+				ssl.AlertSequence++
 			}
 		}
+
+		myAlertValue := reflect.ValueOf(newAlert.Data).Elem()
+		myAlertType := myAlertValue.Type()
+		propAlertType := proto.GetProperties(myAlertType)
+
+		alertTypeName := propAlertType.Prop[0].OrigName
+
+		tags["severity"] = fmt.Sprintf("%v", newAlert.Severity)
+
+		raise := *newAlert.Raised
+
+		switch alertTypeName {
```

Comment:
Using reflection and mapping alertTypeData we can avoid switch and simplify the code. Below is one idea 

If you see the type_name is mapped to GetTypeName. Having a little function that map type_name to GetTypeName and then using reflect we can achieve the functionality.

Reply:
I've implemented this suggestion, but left the switch statements commented out for now until further review - thx.

Reply:
Reverted this back to the switch statement until we have time to profile the reflect based solution.